### ISSUE TITLE
`org-roam-date-filename-format` not working as expected

### ISSUE THREAD

**Issue by Elvecent** [jethrokuan/org-roam#347]
#### Steps to Reproduce
Example:
1. Set `org-roam-date-filename-format` to a value different from `org-roam-date-title-format`
2. Run `org-roam-date` for some date
3. Save the note

#### Expected Results
The filename is formatted with respect to `org-roam-date-filename-format`

#### Actual Results
The filename is formatted with respect to `org-roam-date-title-format`

### Versions
- Emacs (`C-h v emacs-version`): v26.3
- Org-roam commit: the one that made it into MELPA as of now

#### Proposed fix
EDIT: the fix is a little off in that it adds a redundant `.org` in the name.
```elisp
(defun org-roam--file-for-time (time)
  "Create and find file for TIME."
  (let* ((title (format-time-string org-roam-date-title-format time))
         (filename (format-time-string org-roam-date-filename-format time))
         (file-path (org-roam--file-path-from-id filename)))
    (if (file-exists-p file-path)
        (find-file file-path)
      (let ((org-roam-capture-templates (list (list "d" "daily" 'plain (list 'function #'org-roam--capture-get-point)
                                                    ""
                                                    :immediate-finish t
                                                    :file-name "${file-path}" ;; here
                                                    :head "#+TITLE: ${title}")))
            (org-roam--capture-context 'title)
            (org-roam--capture-info (list (cons 'title title) (cons 'file-path file-path)))) ;; and here
        (add-hook 'org-capture-after-finalize-hook #'org-roam--capture-find-file-h)
        (org-roam-capture)))))
```

**Comment by Elvecent** [comment]
Sorry, my fix is not quite right. It ends up generating `.org` extension twice. Not sure if it's a nice solution but I got around that with just cutting it off:
```elisp
(defun org-roam--file-for-time (time)
  "Create and find file for TIME."
  (let* ((title (format-time-string org-roam-date-title-format time))
         (filename (format-time-string org-roam-date-filename-format time))
         (file-path (org-roam--file-path-from-id filename)))
    (if (file-exists-p file-path)
        (find-file file-path)
      (let ((org-roam-capture-templates (list (list "d" "daily" 'plain (list 'function #'org-roam--capture-get-point)
                                                    ""
                                                    :immediate-finish t
                                                    :file-name "${file-path}"
                                                    :head "#+TITLE: ${title}")))
            (org-roam--capture-context 'title)
            (cons 'file-path (replace-regexp-in-string "\\.org$" "" file-path)))))
        (add-hook 'org-capture-after-finalize-hook #'org-roam--capture-find-file-h)
        (org-roam-capture)))))
```

**Comment by jethrokuan** [comment]
I think I fixed it already in #349 , let me know if that one doesn't work.
